.github/workflows/pr-check.yml

```diff
@@ -0,0 +1,35 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: OSV PR format check
+
+on:
+  # `pull_request_target` is only required when editing PRs from forks.
+  pull_request:
+    types:
+      - opened
+      - edited
+      - reopened
+
+permissions:
```

Comment:
I think based on my recent faffing around, you can move this out of the top level and into the specific job that requires it for more precise definition.

Then explicitly set the permissions at this level to `{}`

Reply:
I do see our other Git action yaml file set the permission at each job level. I thought this is just because there are multiple jobs with different permissions required. 

Is there any benefit of setting the PR `read` permission at the job level instead of the top level for PR check workflow? (I am assuming all jobs within pr_check.yml would require PR read permission)